adoc-mode.el

```diff
@@ -369,8 +370,77 @@ customizable.")
 (defvar adoc-mode-abbrev-table nil
   "Abbrev table in use in adoc-mode buffers.")
 
-(defvar adoc-font-lock-keywords nil
-  "Font lock keywords in adoc-mode buffers.")
+(defconst adoc-title-0-regexp
+  (rx bol
+      "="
+      space
+      (one-or-more any)
+      eol)
+  "Regular expression for AsciiDoc Document Title (Level 0).")
+
+(defconst adoc-title-1-regexp
+  (rx bol
+      "=="
+      space
+      (one-or-more any)
+      eol)
+  "Regular expression for AsciiDoc Level 1 Section Title.")
+
+(defconst adoc-title-2-regexp
+  (rx bol
+      "==="
+      space
+      (one-or-more any)
+      eol)
+  "Regular expression for AsciiDoc Level 2 Section Title.")
+
+(defconst adoc-title-3-regexp
+  (rx bol
+      "===="
+      space
+      (one-or-more any)
+      eol)
+  "Regular expression for AsciiDoc Level 3 Section Title.")
+
+(defconst adoc-title-4-regexp
+  (rx bol
+      "===="
+      space
+      (one-or-more any)
+      eol)
+  "Regular expression for AsciiDoc Level 4 Section Title.")
+
+(defconst adoc-title-5-regexp
+  (rx bol
+      "====="
+      space
+      (one-or-more any)
+      eol)
+  "Regular expression for AsciiDoc Level 5 Section Title.")
+
+(defconst adoc-table-regexp
+  (rx (seq
+       (minimal-match
+        bol
+        "|==="
+        eol
+        (zero-or-more anything)
+        bol
+        "|==="
+        eol)))
+  "Regular expression for AsciiDoc table.")
+
+(defconst adoc-code-regexp
+  (rx (seq
+       (minimal-match
+        bol
+        "----"
```

Comment:
This rigid kind of regexp is wrong.
The fence is actually `"----+" and the end fence must exactly match the start fence.
This makes nested blocks possible.